lib/pbench/server/api/resources/query_apis/search_api.py

```diff
@@ -0,0 +1,201 @@
+from flask import jsonify
+from logging import Logger
+from typing import Any, AnyStr, Dict
+
+from pbench.server import PbenchServerConfig
+from pbench.server.api.resources.query_apis import (
+    ElasticBase,
+    Schema,
+    Parameter,
+    ParamType,
+    PostprocessError,
+)
+
+
+class SearchAPI(ElasticBase):
+    """
+    Create a search request based on given query parameter.
+    """
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        super().__init__(
+            config,
+            logger,
+            Schema(
+                Parameter("user", ParamType.USER, required=False),
+                Parameter("start", ParamType.DATE, required=True),
+                Parameter("end", ParamType.DATE, required=True),
+                Parameter("search_term", ParamType.STRING, required=False),
+                Parameter("fields", ParamType.LIST, required=False),
+            ),
+        )
+
+    def flatten_json(
+        self, es_json: Dict[AnyStr, Any], prefix="", out={}
+    ) -> Dict[AnyStr, Any]:
+        """
+        Utility function for flattening the json.
+        json_in =
```

Comment:
A bit more description would be useful; for a start, explain what "flattening" means. While it's fairly obvious if you read the examples below closely, a quick summary would be nice. (And particularly mention your renaming algorithm to deal with duplicates.)

Add a blank line to separate the examples more clearly. And, in between, a semantic description of the parameters, which are not obvious to a casual reader.

(The type hints will be helped by the `JSON` typedef Webb and I worked out, but of course that's not in yet.)